smartgrid-prediction/README.asciidoc

```diff
@@ -0,0 +1,128 @@
+# Introduction
+
+This sample showcases a fictional use of Spring XD in the context of smart electricity metering and production. It was inspired by the http://www.cse.iitb.ac.in/debs2014/?page_id=42[ACM Distributed Event Based Systems Grand Challenge 2014] and re-uses some of its sample input data.
+
+Smart plugs are used to measure power consumption and report metrics to the system. In this sample, smartplugs are simulated by data ingestion using JMeter. Also, to have the demo display some lively data, everything has been re-sampled to be played at 60x the original speed (as such, the 16hrs worth of data available are ingested in ~16mins).
+
+The ingestion, aggregation and machine learning features of Spring XD are used to display and predict the energy consumption of the grid, in a scalable way.
+
+The overall logical architecture of this sample is depicted below:
+
+image::arch.png[]
+
+# Understanding the sample
+
+You'll find several directories in this sample. Not all are needed to actually run the solution, but are included for complete reference. In addition, the main piece of the puzzle is the list of stream definitions that make up the XD solution. You'll read about them in the xref:running[] section.
+
+* link:jmeter[] contains the JMeter plan and supporting scripts that can be used to ingest data in Spring XD.
+* link:frontend[] contains a small serverside app (and accompanying frontend) that reads aggregated counters and displays them
+* link:prediction[] is the home of the link:prediction/metaModel.pmml[Neural Network models] that can predict consumption, as well as explanations about how to re-create that file from scratch.
+
+[[running]]
+# Running the demo
+## Install the analytic-pmml module
```

Comment:
@ericbottard This needs to be rewritten. The module is already included in XD. Therefore running `setup-pmml.sh` yields: `There is already a module named 'analytic-pmml' with type 'processor'`.